tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts

```diff
@@ -86,16 +67,15 @@ export class Im2ColProgram implements WebGPUProgram {
             int blockIndex = rc[0];
             int pos = rc[1];
 
-            int offsetY = int(blockIndex / ${outWidth}) * ${strideHeight} -
-              ${top};
-            int d0 = offsetY + ${dilationHeight} * (pos / ${itemsPerBlockRow});
+            int offsetY = int(blockIndex / outWidth) * stride[1] - pad[1];
+            int d0 = offsetY + dilation[1] * (pos / itemsPerBlockRow);
             float value = 0.0;
             if(d0 < aShape[${rowDim}] && d0 >= 0) {
-              int offsetX = int(mod(float(blockIndex), ${outWidth}.) *
-                ${strideWidth}. - ${left}.);
-              int d1 = offsetX + ${dilationWidth} * (int(mod(float(pos),
-                ${itemsPerBlockRow}.) / ${inChannels}.));
-              int ch = int(mod(float(pos), ${inChannels}.));
+              int offsetX = int(mod(float(blockIndex), float(outWidth)) *
+              float(stride[0]) - float(pad[0]));
```

Comment:
Why do we need to convert it to `float`? Why not directly use `int offsetX = mod(blockIndex, outWidth) * stride[0] - pad[0];`?

Reply:
Remove float conversion is done